lib/spack/spack/relocate.py

```diff
@@ -193,19 +198,34 @@ def get_filetype(path_name):
     file = Executable('file')
     file.add_default_env('LC_ALL', 'C')
     output = file('-b', '-h', '%s' % path_name,
-                  output=str, err=str)
+                  output=str, error=str)
     return output.strip()
 
 
-def modify_elf_object(path_name, orig_rpath, new_rpath):
+def strings_contains_installroot(path_name):
+    """
+    Check if the file contain the install root string.
+    """
+    strings = Executable('strings')
+    output = strings('%s' % path_name,
+                     output=str, err=str)
+    return (spack.store.layout.root in output)
+
+
+def modify_elf_object(path_name, new_rpaths):
     """
     Replace orig_rpath with new_rpath in RPATH of elf object path_name
     """
     if platform.system() == 'Linux':
-        new_joined = ':'.join(new_rpath)
+        new_joined = ':'.join(new_rpaths)
         patchelf = Executable(get_patchelf())
-        patchelf('--force-rpath', '--set-rpath', '%s' % new_joined,
-                 '%s' % path_name, output=str, cmd=str)
+        try:
+            patchelf('--force-rpath', '--set-rpath', '%s' % new_joined,
+                     '%s' % path_name, output=str, error=str)
+        except ProcessError as e:
+            tty.die('patchelf --set-rpath %s failed' %
+                     path_name, e)
```

Comment:
@gartung I think you need to move this one column to the left to pass the flake check